src/modules/jcms_admin/src/Plugin/views/field/ContentLink.php

```diff
@@ -0,0 +1,145 @@
+   */
+  public function query() {
+    // Leave empty to avoid a query on this field.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $values) {
+    /** @var \Drupal\node\NodeInterface $content */
+    $content = $values->_entity;
+    $journal = \Drupal::getContainer()->hasParameter('journal') ? \Drupal::getContainer()->getParameter('journal') : [];
+    $path = $this->getContentPath($content);
+    if (!$path || !$journal || !isset($journal['path'])) {
+      return '';
+    }
+    else {
+      $base_path = $journal['path'];
+      $render = [
+        [
+          '#theme' => 'clipboard_simple',
+          '#text' => $base_path . $path,
+        ],
+      ];
+      if ($preview_path = $journal['preview']) {
+        $render[] = [
+          '#theme' => 'clipboard_simple',
+          '#text' => $preview_path . $path,
+          '#button_text' => 'Copy preview url',
+        ];
+      }
+      return $render;
+    }
+  }
+
+  /**
+   * Get content path.
+   */
+  private function getContentPath(NodeInterface $content) {
+    $whitelist = [
+      'annual_report',
+      'blog_article',
+      'collection',
+      'event',
+      'interview',
+      'labs_experiment',
+      'podcast_episode',
+    ];
+    $sns = \Drupal::service('jcms_notifications.entity_crud_notification_service');
+    /** @var \Drupal\jcms_notifications\Notification\BusOutgoingMessage $message */
+    if (in_array($content->bundle(), $whitelist) && $message = $sns->getMessageFromEntity($content)) {
+      $stem = $message->getTopic();
+      $id = $message->getId();
+      $slug = NULL;
+      $seperator = '/';
+      switch ($content->bundle()) {
+        case 'annual_report':
+          $id = NULL;
+          break;
+
+        case 'blog_article':
+          $stem = 'inside-elife';
+          $slug = \Drupal::service('slugify')->slugify($content->getTitle());
+          break;
+
+        case 'interview':
+          $slug = \Drupal::service('slugify')->slugify($content->get('field_person_preferred_name')->getString());
+          break;
+
+        case 'labs_experiment':
+          $stem = 'labs';
+          $slug = \Drupal::service('slugify')->slugify($content->getTitle());
+          break;
+
+        case 'podcast_episode':
+          $stem = 'podcast/episode';
+          $seperator = '';
+          break;
+
+        default:
+          $slug = \Drupal::service('slugify')->slugify($content->getTitle());
+      }
+
+      return sprintf('/%s', implode($seperator, array_filter([$stem, $id, $slug])));
+    }
+    elseif ($content->bundle() === 'article') {
+      return sprintf('/%s/%s', 'articles', $content->getTitle());
+    }
+    elseif ($content->bundle() === 'highlight_list') {
+      return sprintf('/%s', $content->getTitle());
+    }
+    elseif ($content->bundle() === 'job_advert') {
+      return sprintf('/%s/%s/%s', 'jobs', substr($content->uuid(), -8), \Drupal::service('slugify')->slugify($content->getTitle()));
```

Comment:
`substr($content->uuid(), -8)` is duplication that can be extracted as a private method

Reply:
See also ELPP-3070.